miner/worker.go

```diff
@@ -991,10 +985,30 @@ func (w *worker) commit(uncles []*types.Header, interval func(), update bool, st
 	return nil
 }
 
+// deep copy current receipts. This is should be used whene an asynchronus process can modify
+// current context.
+func (w *worker) copyCurrentReceipts() []*types.Receipt {
+	receipts := make([]*types.Receipt, len(w.current.receipts))
+	for i, l := range w.current.receipts {
+		cpy := *l
+		receipts[i] = &cpy
+	}
+	return receipts
+}
```

Comment:
IMHO moving this method out doesn't help much. It was 5 lines in total, a trivial slice copy, that we don't even use in another place to warrant separation.

Reply:
Look at  the usage:
```
receipts := w.copyCurrentReceipts()
```
vs 
```
	receipts := make([]*types.Receipt, len(w.current.receipts))
	for i, l := range w.current.receipts {
		cpy := *l
		receipts[i] = &cpy
	}
```


If you read functions, it's good to have this kind of abstraction. You don't need to focus on parsing the loop, and you quickly gasp that you just want to obtain a copy. 

Side comment: this file has lot of lengthy functions and in many places it's hard to follow it.